### PR TITLE
Angular 11 Upgrade

### PR DESCRIPTION
**Pull request by olgapuig**
Issue: #413

Angular 11 Upgrade
Guide: https://update.angular.io/?l=3&v=10.2-11.0

Primeng 11 Upgrade
Guide: https://github.com/primefaces/primeng/wiki/Migration-Guide
Remove the locale property in the date picker components to use i18n

Use 'karma-coverage' because 'karma-coverage-istanbul-reporter' usage has been deprecated since version 11. 
The change in the  karma.conf.js is also aligned with the new schematics https://github.com/angular/angular-cli/blob/v11.0.0/packages/schematics/angular/application/files/karma.conf.js.template

Use systelab-translate and systelab-preferentes in Angular11

Update to Typescript 4
There's a breaking change in which the "Properties Overriding Accessors (and vice versa) is an Error". Some combos in Modulab/HemoHub overwrite the startsWith property as @Input, so it must included in the abstract combo.

Update Browserslist and dependencies with the following command to fix the issue "[error] BrowserslistError: Unknown version 88 of edge"
npx browserslist@latest --update-db

### CODE REVIEW

File: projects/systelab-components/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (carlosra85) [inline projects/systelab-components/package.json:1]: PrimeNg should be updated to 11 to be compatible?

**Review by carlosra85 — COMMENTED** [review]
Before publishing remember to publish a branch 11.0.x or 11.1.x to future changes in versions related to that versions of components and Angular 10

File: projects/systelab-components/package.json (hunk 1 above)
Comment (olgapuig) [reply projects/systelab-components/package.json:1]: Ok, I will review it.

**Review by vinyulis — COMMENTED** [review]
Before systelab-components migration systelab-preferences and systelab-tranlate shoud be migrated to Angular 11 and published, systelab-components uses both

**Review by alfonsserra — COMMENTED** [review]
Should we upgrade translate and preferences first?

**Comment by olgapuig** [comment]
> Should we upgrade translate and preferences first?
Sure, I will do it.

**Comment by olgapuig** [comment]
> Before publishing remember to publish a branch 11.0.x or 11.1.x to future changes in versions related to that versions of components and Angular 10

The latest branch is the "10.3.x". I will be creating 11.0.x once this PR will be approved.

**Comment by olgapuig** [comment]
Branch 11.0.x is created with the master for future changes in Angular 10

**Review by carlosra85 — DISMISSED** [review]

**Review by RubKrmon — APPROVED** [review]

**Review by carlosra85 — DISMISSED** [review]

**Review by RubKrmon — DISMISSED** [review]

**Review by vinyulis — COMMENTED** [review]
Where is protected config: PrimeNGConfig in datepickers used?. It is going to be used by applications that use this component.?

**Comment by olgapuig** [comment]
> Where is protected config: PrimeNGConfig in datepickers used?. It is going to be used by applications that use this component.?

This injection is not needed into the datepickers. I already tested in the angular11 branch in the source code of HemoHub/Modulab and the internationalization worked well. So I will change it to private. Thanks for the feedback!

**Review by vinyulis — APPROVED** [review]
OK!! Thanks!!

**Review by carlosra85 — APPROVED** [review]

**Review by RubKrmon — APPROVED** [review]

**Pull request merged**